src/eddymotion/math/utils.py

```diff
@@ -0,0 +1,51 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2024 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY kIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+import numpy as np
+
+
+def compute_angle(v1, v2, smallest=False):
+    """Compute the angle between two vectors.
+
+    Parameters
+    ----------
+    v1 : :obj:`ndarray`
+        First vector.
+    v2 : :obj:`ndarray`
+        Second vector.
+    smallest : :obj:`bool`
+        ``True`` to compute the smallest angle between the two vectors.
```

Comment:
`smallest` feels like a misnomer - This is just ensuring the maximum angle is 90 deg. by flipping the direction of vectors with larger angles, correct?

Reply:
Agreed; have not been sharp enough to come up with a better name following the description in the paper. Would be grateful if you could come up a suggestion.

Reply:
```suggestion
    closest_polarity : :obj:`bool`
        If ``True``, the polarity of ``v2`` is reversed if the angle w.r.t. ``v1`` is smaller.
```